plugins/dominant-color-images/class-dominant-color-image-editor-gd.php

```diff
@@ -78,9 +78,6 @@ public function has_transparency() {
 					return new WP_Error( 'unable_to_obtain_rgb_via_imagecolorat' );
 				}
 				$rgba = imagecolorsforindex( $this->image, $rgb );
-				if ( ! is_array( $rgba ) ) {
-					return new WP_Error( 'unable_to_obtain_rgba_via_imagecolorsforindex' );
-				}
```

Comment:
PHP docs say that this function only returns an array: https://www.php.net/manual/en/function.imagecolorsforindex.php

But PhpStorm's stubs say it can return `false`:

```php
/**
 * Get the colors for an index
 * @link https://php.net/manual/en/function.imagecolorsforindex.php
 * @param resource|GdImage $image
 * @param int $color <p>
 * The color index.
 * </p>
 * @return array|false an associative array with red, green, blue and alpha keys that
 * contain the appropriate values for the specified color index or <b>FALSE</b> on failure
 */
#[Pure]
#[LanguageLevelTypeAware(['8.0' => 'array'], default: 'array|false')]
#[ArrayShape(["red" => "int", "green" => "int", "blue" => "int", "alpha" => "int"])]
function imagecolorsforindex(GdImage $image, int $color) {}
```

Again, it seems PhpStorm is wrong.

Reply:
According to the [changelog](https://www.php.net/manual/en/function.imagecolorsforindex.php#refsect1-function.imagecolorsforindex-changelog):

> imagecolorsforindex() now throws a [ValueError](https://www.php.net/manual/en/class.valueerror.php) exception if color is out of range; previously, [false](https://www.php.net/manual/en/reserved.constants.php#constant.false) was returned instead.

Reply:
Ah, then we should probably wrap this in a try/catch block to account for this.